### PR TITLE
Added a flag on takeWhile to allow it to include the final element, #21330

### PR DESCRIPTION
**Pull request by patriknw**
Completing https://github.com/akka/akka/pull/20374
I just added the mima filter

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM, needs a rebase sadly though

File: akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala [hunk 1]
```diff
@@ -592,12 +611,13 @@ class SubFlow[-In, +Out, +Mat](delegate: scaladsl.SubFlow[Out, Mat, scaladsl.Flo
...
    * '''Backpressures when''' downstream backpressures
    *
-   * '''Completes when''' predicate returned false or upstream completes
+   * '''Completes when''' predicate returned false (or 1 after predicate returns false if `inclusive` or upstream completes
    *
    * '''Cancels when''' predicate returned false or downstream cancels
    */
-  def takeWhile(p: function.Predicate[Out]): SubFlow[In, Out, Mat] =
-    new SubFlow(delegate.takeWhile(p.test))
+  // TODO: Merge these methods when it's ok to break binary compatibility
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/SubFlow.scala]: remove todo? Don't think it's a problem right?

**Comment by patriknw** [comment]
rebased and removed todo

**Comment by patriknw** [comment]
Refs #21330

**Pull request merged**